msal/src/main/java/com/microsoft/identity/client/AuthenticationResult.java

```diff
@@ -32,29 +32,52 @@
 import java.util.List;
 import java.util.concurrent.TimeUnit;
 
+import static com.microsoft.identity.common.internal.authscheme.PopAuthenticationSchemeInternal.SCHEME_POP;
+
 /**
  * MSAL successful authentication result. When auth succeeds, token will be wrapped into the
  * {@link AuthenticationResult} and passed back through the {@link AuthenticationCallback}.
  */
 public final class AuthenticationResult implements IAuthenticationResult {
 
+    private static final String SCHEME_DELIMITER = " ";
+
     private final String mTenantId;
     private final AccessTokenRecord mAccessToken;
     private final IAccount mAccount;
+    private final String mAuthorizationHeader;
 
-    public AuthenticationResult(@NonNull final List<ICacheRecord> cacheRecords) {
+    AuthenticationResult(@NonNull final List<ICacheRecord> cacheRecords,
+                         @NonNull final String authHeader) {
         final ICacheRecord mostRecentlyAuthorized = cacheRecords.get(0);
         mAccessToken = mostRecentlyAuthorized.getAccessToken();
         mTenantId = mostRecentlyAuthorized.getAccount().getRealm();
         mAccount = AccountAdapter.adapt(cacheRecords).get(0);
+        mAuthorizationHeader = authHeader;
     }
 
     @Override
     @NonNull
     public String getAccessToken() {
+        if (mAuthorizationHeader.startsWith(SCHEME_POP + SCHEME_DELIMITER)) {
```

Comment:
why do we have to do this for PoP? 

I'd expect mAccessToken.getSecret() to return the same value. Isn't that the case?